documentation/src/main/asciidoc/server_guide/server_guide.asciidoc

```diff
@@ -1002,3 +1002,99 @@ remoteCacheManager = new RemoteCacheManager(clientBuilder.build());
 
 Additionally, you might also want to enable client certificate authentication (and therefore also allow the use of the EXTERNAL SASL mech to authenticate clients).
 
+== Health monitoring
+
+Infinispan server has special endpoints for monitoring cluster health. The API is exposed via:
+
+* Programmatically (using `embeddedCacheManager.getHealth()`)
+* JMX
+* CLI
+* REST (using https://docs.jboss.org/author/display/WFLY10/The+HTTP+management+API[WildFly HTTP Management API])
+
+=== Accessing Health API using JMX
+
+At first you need to connect to the Infinispan Server using JMX (use JConsole or other tool for this).
+Next, navigate to object name `jboss.datagrid-infinispan:type=CacheManager,name="clustered",component=CacheContainerHealth`.
+
+=== Accessing Health API using CLI
+
+The Health API can be also accessed using CLI interface. An example invocation and result is presented below:
+
+[source,bash]
+----
+$ ispn-cli.sh -c "/subsystem=datagrid-infinispan/cache-container=clustered/health=HEALTH:read-resource(include-runtime=true)"
+
+{
+    "outcome" => "success",
+    "result" => {
+        "cache-health" => "GREEN",
+        "cluster-health" => ["test"],
```

Comment:
Sorry, you are absolutely correct... the output should be `"cluster-health" : ["test", "GREEN"]` since we use `RED`, `YELLOW` and `GREEN` (see the [design](https://github.com/infinispan/infinispan/wiki/Health-check-API)).